### PR TITLE
Bug fixes from previous version of orientation module

### PR DESCRIPTION
**Pull request by saurabhmittal23**
- Works for both orientation formats (long, lat) and (lat, long)

### CODE REVIEW

**Comment by hiyoneda** [comment]
I confirmed that the bugs were solved. 

I have a comment on get_l_b(self, pointing, which). Here, the order of latitude and longitude in the array are judged by analyzing it, e.g., checking the range of the 1st and 2nd columns. However, it may cause errors in some cases, for instance, when one uses an orientation file in a very short time interval ( both (l,b) can be within -90 to +90 degrees).

Can you specify the order of (l, b) as an input parameter? Then, you can check whether the range of parameters is consistent with the given order of (l, b).

**Comment by Yong2Sheng** [comment]
> I confirmed that the bugs were solved.
> 
> I have a comment on get_l_b(self, pointing, which). Here, the order of latitude and longitude in the array are judged by analyzing it, e.g., checking the range of the 1st and 2nd columns. However, it may cause errors in some cases, for instance, when one uses an orientation file in a very short time interval ( both (l,b) can be within -90 to +90 degrees).
> 
> Can you specify the order of (l, b) as an input parameter? Then, you can check whether the range of parameters is consistent with the given order of (l, b).

Hi Hiroki! Thank you for checking it out!

The default order is (l, b), if both columns meet the criteria, it will read the columns with the order of (l, b). 

We can add that parameter assuming the users have the knowledge of the order of input. Then get_l_b function can double check that and correct it if the order doesn't match the order input from the user, and print out an warning.

What do you think?

**Comment by hiyoneda** [comment]
(sorry for my delayed response).

Yes, I like your idea! I think that it is good that we specify the order of (l, b) when we know it in advance, which would avoid unexpected errors.

**Review by israelmcmc — CHANGES_REQUESTED** [review]
Thanks @saurabhmittal23 and @Yong2Sheng for working on fixing bug. If I understand correctly, the issue is that the users need to pass an array of shape (2, N), and sometimes the user happens to use (l,b) and sometimes (b,l). You can correct me if I'm wrong, but I got it right, my opinion is that this is not the best way to go. 

Object-oriented programming allows you to pass instances of well-defined classes, instead of raw data, which makes the code easier to maintain and more flexible in the long run. In this case, instead of passing a raw array, and then converting to SkyCoord object, I think the user needs to pass SkyCoord directly (just like scoords). The instances of SkyCoord classes contain the reference frame, can be transformed, and the order of the coordinates (e.g. l,b) becomes a non-issue. This makes the code less error prone. 

Changing this means that the methods that read from files need to create SkyCoord objected, but that's fine, because the file formats are assumed to be well--defined and known.

File: cosipy/threeml/COSILike.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 from astromodels import Parameter
 
 from cosipy.response.FullDetectorResponse import FullDetectorResponse
-from cosipy.coordinates.orientation import Orientation_file
+#from cosipy.coordinates.orientation import Orientation_file
```
[the remark this answers is not in the document]
Comment (Yong2Sheng) [reply cosipy/threeml/COSILike.py]: Removed

**Comment by Yong2Sheng** [comment]
I've changed the inputs of SpacecraftFile from raw data to objects. Please check **docs/tutorials/Point_source_resonse.ipynb** for the usage. Thanks!

**Comment by hiyoneda** [comment]
Can you modify "parse_from_file"? Now, the variables when creating the instance of SpacecraftFile are the set of objects of  astropy.time.Time, SkyCoord. However, in "parse_from_file", numpy arrays are used, which causes an error.

**Review by israelmcmc — APPROVED** [review]

**Comment by israelmcmc** [comment]
Thank @saurabhmittal23 , @Yong2Sheng  and @hirokiyoneda . Looks good to me, I'll merge

**Pull request merged**